lib/standards.h

```diff
@@ -38,10 +38,10 @@ struct Standards {
     enum Language : std::uint8_t { None, C, CPP };
 
     /** C code standard */
-    enum cstd_t : std::uint8_t { C89, C99, C11, C17, C23, CLatest = C23 } c = CLatest;
+    enum cstd_t : std::uint8_t { C89, C99, C11, C17, C23, CLatest = C23, CInvalid } c = CLatest;
 
     /** C++ code standard */
-    enum cppstd_t : std::uint8_t { CPP03, CPP11, CPP14, CPP17, CPP20, CPP23, CPP26, CPPLatest = CPP26 } cpp = CPPLatest;
+    enum cppstd_t : std::uint8_t { CPP03, CPP11, CPP14, CPP17, CPP20, CPP23, CPP26, CPPLatest = CPP26, CPPInvalid } cpp = CPPLatest;
```

Comment:
hmm an alternative is that you remove the CPPInvalid enum constant and throw exceptions in getC and getCpp if invalid strings are provided.. with this approach we need to handle the CPPInvalid and CInvalid constants in more places.

I believe the CI failure indicates that the --project does not handle CppInvalid well.

That exception should be handled both for "--std" arguments and "--project".

Reply:
The code that handles '--project' doesn't seem to call Standards::getC/getCPP (please correct me if I'm wrong). Maybe I could do "dummy" calls to Standards::getC/getCPP in ImportObject::fsParseCommands in order to throw and catch on any unknown --std strings in the cmd line parsing?

Reply:
ehm.. I am starting to feel that maybe exceptions are not required. but cppinvalid is not required neither. there are different options and I don't have a strong opinion at this point.

> please correct me if I'm wrong
See importproject:
```
daniel@laptop:~/cppcheck$ git grep "getCPP[(][^)]"
cli/cmdlineparser.cpp:                    mSettings.standards.cpp = Standards::getCPP(std);
lib/importproject.cpp:                fs.standard = Standards::getCPP(i.cppstd);
lib/preprocessor.cpp:        splitcfg(mSettings.platform.getLimitsDefines(Standards::getCPP(dui.std)), dui.defines, "");
lib/standards.h:        cpp = getCPP(str);
lib/standards.h:        return getCPP(cpp);
lib/standards.h:    static std::string getCPP(cppstd_t std) {
lib/standards.h:    static cppstd_t getCPP(const std::string &std) {
```